plugins/woocommerce/client/blocks/assets/js/blocks/mini-cart/mini-cart-contents/inner-blocks/mini-cart-footer-block/block.tsx

```diff
@@ -42,13 +42,19 @@ const Block = ( {
 	className,
 	cartButtonLabel,
 	checkoutButtonLabel,
-}: Props ): JSX.Element => {
+}: Props ): JSX.Element | null => {
 	const { cartTotals } = useStoreCart();
 	const subTotal = getSetting( 'displayCartPricesIncludingTax', false )
 		? parseInt( cartTotals.total_items, 10 ) +
 		  parseInt( cartTotals.total_items_tax, 10 )
 		: parseInt( cartTotals.total_items, 10 );
 
+	// If the block is hidden, then the children will be undefined.
+	// In this case, we return an empty fragment to prevent the fallback for the legacy template
+	if ( children === undefined ) {
```

Comment:
Would it be better to check if children is falsy rather than explicitly `undefined` that way we catch empty arrays and null, for example.